Tests/scripts/download_conf_repos.sh

```diff
@@ -1,11 +1,53 @@
 #!/usr/bin/env bash
 set +e
 
-# replace slashes ('/') in the branch name, if exist, with underscores ('_')
+clone_repository() {
+  local repo_name=$1
+  local branch=$2
+  local retry_count=$3
+  local exit_code=0
+  local i=1
+  echo "clone ${repo_name} from branch: ${branch} with ${retry_count} retries"
+  for ((i=1; i <= retry_count; i++)); do
+    git clone --depth=1 "https://gitlab-ci-token:${CI_JOB_TOKEN}@${repo_name}.git" --branch "${branch}" && exit_code=0 && break || exit_code=$?
+    if [ ${i} -ne "${retry_count}" ]; then
+      echo "Failed to clone ${repo_name} with branch:${branch}, exit code:${exit_code}, sleeping for 10 seconds and trying again"
+      sleep 10
+    else
+      echo "Failed to clone ${repo_name} with branch:${branch}, exit code:${exit_code}, exhausted all retries"
+      break
+    fi
+  done
+  if [ ${exit_code} -ne 0 ]; then
+    branch="master"
+    echo "Failed to clone ${repo_name} with branch:${branch}, will try to clone with branch:${branch}"
```

Comment:
Typo? This `branch` overrides the `branch` from above, and is used twice in the `echo`.
Also further down.
```suggestion
    master_branch="master"
    echo "Failed to clone ${repo_name} with branch ${branch}, will try to clone with branch ${master_branch}"
```